src/lib/storage/table.cpp

```diff
@@ -16,54 +16,66 @@
 #include "utils/assert.hpp"
 
 namespace opossum {
+Table::Table(uint32_t chunk_size) : _max_chunk_size(chunk_size) { _chunks.push_back(std::make_shared<Chunk>()); }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(row_count() == 0, "The table already contains data so adding a column is not possible.");
+
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+
+  auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(type);
+
+  _chunks.front()->add_segment(segment);
 }
 
 void Table::append(std::vector<AllTypeVariant> values) {
-  // Implementation goes here
-}
+  DebugAssert(values.size() == column_count(), "The number of values doesn't match number of columns.");
 
-uint16_t Table::column_count() const {
-  // Implementation goes here
-  return 0;
+  if (_chunks.back()->size() >= _max_chunk_size) {
```

Comment:
if _chunks.back()->size() > _max_chunk_size, we would have some problem, so I would not assume that here.

Reply:
I went with Ivans version :)